fake_joint_driver/src/fake_joint_driver.cpp

```diff
@@ -25,103 +24,145 @@ FakeJointDriver::FakeJointDriver(void)
   }
   // Include joints into joint_set
-  for (auto i=0; i< include_joints_.size(); i++)
-  {
+  for (auto i = 0; i < include_joints_.size(); i++) {
     joint_set.insert(include_joints_[i]);
   }
   // Exclude joints in joint_set
-  for (auto i=0; i< exclude_joints_.size(); i++)
-  {
+  for (auto i = 0; i < exclude_joints_.size(); i++) {
     joint_set.erase(exclude_joints_[i]);
   }
   // Convert to vector (joint_names_)
-  std::copy(joint_set.begin(), joint_set.end(), std::back_inserter(joint_names_));
+  std::copy(joint_set.begin(), joint_set.end(),
+            std::back_inserter(joint_names_));
   // Check the emptyness of joints
   if (joint_names_.size() == 0) {
     ROS_ERROR("No joints is specified. Please use include_joints parameters.");
     ros::shutdown();
   }
   // Resize members
   cmd_dis.resize(joint_names_.size());
+  cmd_dis_old.resize(joint_names_.size());
+  cmd_vel.resize(joint_names_.size(), 0.0);
   act_dis.resize(joint_names_.size());
   act_vel.resize(joint_names_.size());
   act_eff.resize(joint_names_.size());
 
   // Set start position
-  for (auto it=start_position_map.begin(); it!=start_position_map.end(); it++)
-  {
-    for (auto i=0; i<joint_names_.size(); i++)
-    {
-      if (joint_names_[i] == it->first)
-      {
+  for (auto it = start_position_map.begin(); it != start_position_map.end();
+       it++) {
+    for (auto i = 0; i < joint_names_.size(); i++) {
+      if (joint_names_[i] == it->first) {
         act_dis[i] = it->second;
         cmd_dis[i] = it->second;
+        cmd_dis_old[i] = cmd_dis[i];
       }
     }
   }
 
   // Create joint_state_interface and position_joint_interface
-  for (int i = 0; i< joint_names_.size(); i++)
-  {
+  for (int i = 0; i < joint_names_.size(); i++) {
     ROS_DEBUG_STREAM("joint[" << i << "]:" << joint_names_[i]);
     // Connect and register the joint_state_interface
-    hardware_interface::JointStateHandle state_handle(joint_names_[i], &act_dis[i], &act_vel[i], &act_eff[i]);
+    hardware_interface::JointStateHandle state_handle(
+        joint_names_[i], &act_dis[i], &act_vel[i], &act_eff[i]);
     joint_state_interface.registerHandle(state_handle);
 
     // Connect and register the position_joint_interface
-    hardware_interface::JointHandle pos_handle(joint_state_interface.getHandle(joint_names_[i]), &cmd_dis[i]);
+    hardware_interface::JointHandle pos_handle(
+        joint_state_interface.getHandle(joint_names_[i]), &cmd_dis[i]);
     position_joint_interface.registerHandle(pos_handle);
+
+    // Connect and register the velocity_joint_interface
+    hardware_interface::JointHandle vel_handle(
+        joint_state_interface.getHandle(joint_names_[i]), &cmd_vel[i]);
+    velocity_joint_interface.registerHandle(vel_handle);
   }
   registerInterface(&joint_state_interface);
   registerInterface(&position_joint_interface);
+  registerInterface(&velocity_joint_interface);
 }
 
-FakeJointDriver::~FakeJointDriver()
-{
+FakeJointDriver::~FakeJointDriver() {}
```

Comment:
empty destructor doesn't need to be specified, I'm pretty sure. Go ahead and delete it

Reply:
https://github.com/PickNikRobotics/fake_joint/pull/1/commits/a1941184a48cb560eab2f970460d4cb4d587a669